coll/coll.go

```diff
@@ -370,3 +371,20 @@ func KeyValToMap(s string) (map[string]string, error) {
 	}
 	return m, nil
 }
+
+func MatchLabel(labels map[string]any, key, valuePattern string) bool {
+	for k, v := range labels {
+		if k != key {
+			continue
+		}
+
+		if vStr, ok := v.(string); ok {
+			if collections.MatchItems(vStr, valuePattern) {
+				return true
+			}
+		}
+
+	}
+
+	return false
```

Comment:
I think we need to handle this case of `!*` or just '!` here

Reply:
https://github.com/flanksource/commons/pull/260

Reply:
! or !* should return true if the label doesn't exist

Reply:
https://github.com/flanksource/gomplate/pull/122/files#diff-618040673bae9817053e7d6412d4b1202c80be6dc02c886720c43d53aa74e40bR628-R632